v1_14_4_R1/src/main/java/com/lishid/orebfuscator/nms/v1_14_2_R1/NmsManager.java

```diff
@@ -0,0 +1,617 @@
+			player2.getHandle().playerConnection.sendPacket(packet);
+		}
+	}
+
+	public void notifyBlockChange(World world, IBlockInfo blockInfo) {
+		BlockPosition blockPosition = new BlockPosition(blockInfo.getX(), blockInfo.getY(), blockInfo.getZ());
+		IBlockData blockData = ((BlockInfo)blockInfo).getBlockData();
+
+		((CraftWorld)world).getHandle().notify(blockPosition, blockData, blockData, 0);
+	}
+
+	public int getBlockLightLevel(World world, int x, int y, int z) {
+		return ((CraftWorld)world).getHandle().getLightLevel(new BlockPosition(x, y, z));
+	}
+
+	public IBlockInfo getBlockInfo(World world, int x, int y, int z) {
+		IBlockData blockData = getBlockData(world, x, y, z, false);
+
+		return blockData != null
+				? new BlockInfo(x, y, z, blockData)
+				: null;
+	}
+
+	public int loadChunkAndGetBlockId(World world, int x, int y, int z) {
+		IBlockData blockData = getBlockData(world, x, y, z, true);
+		return blockData != null ? Block.getCombinedId(blockData): -1;
+	}
+
+	public String getTextFromChatComponent(String json) {
+		IChatBaseComponent component = IChatBaseComponent.ChatSerializer.a(json);
+		return CraftChatMessage.fromComponent(component);
+	}
+
+	public boolean isHoe(Material item) {
+		return item == Material.WOODEN_HOE
+				|| item == Material.IRON_HOE
+				|| item == Material.GOLDEN_HOE
+				|| item == Material.DIAMOND_HOE;
+	}
+
+	public boolean isSign(int combinedBlockId) {
+		return BLOCK_ID_SIGNS.contains(combinedBlockId);
+	}
+
+	public boolean isAir(int combinedBlockId) {
+		return BLOCK_ID_AIRS.contains(combinedBlockId);
+	}
+
+	public boolean isTileEntity(int combinedBlockId) {
+		return Block.getByCombinedId(combinedBlockId).getBlock().isTileEntity();
+	}
+
+	public int getCaveAirBlockId() {
+		return BLOCK_ID_CAVE_AIR;
+	}
+
+	public int getBitsPerBlock() {
+		return BITS_PER_BLOCK;
+	}
+
+	public boolean canApplyPhysics(Material blockMaterial) {
+		return blockMaterial == Material.AIR
+				|| blockMaterial == Material.CAVE_AIR
+				|| blockMaterial == Material.VOID_AIR
+				|| blockMaterial == Material.FIRE
+				|| blockMaterial == Material.WATER
+				|| blockMaterial == Material.LAVA;
+	}
+
+	public Set<Integer> getMaterialIds(Material material) {
+		return this.materialIds.get(material);
+	}
+
+	public boolean sendBlockChange(Player player, Location blockLocation) {
+		IBlockData blockData = getBlockData(blockLocation.getWorld(), blockLocation.getBlockX(), blockLocation.getBlockY(), blockLocation.getBlockZ(), false);
+
+		if(blockData == null) return false;
+
+		CraftBlockData craftBlockData = CraftBlockData.fromData(blockData);
+
+		player.sendBlockChange(blockLocation, craftBlockData);
+
+		return true;
+	}
+
+	private static IBlockData getBlockData(World world, int x, int y, int z, boolean loadChunk) {
+		int chunkX = x >> 4;
+		int chunkZ = z >> 4;
+
+		WorldServer worldServer = ((CraftWorld)world).getHandle();
+		// like in ChunkCache, NMS change without R increment.
```

Comment:
will have to investigate bukkit here to see which of the following Methods needs to remain